### DISCUSSION TITLE
SqlClient integration in ASP.NET (.NET 4.8) application - no spans generated

### DISCUSSION THREAD

**Discussion by nixxcz** [getsentry/sentry-dotnet#1967] [Q&A] [answered]
Hello,

I'm running a .NET 4.8 ASP.NET MVC application and I'd like to use the built-in SqlClient integration described [here](https://docs.sentry.io/platforms/dotnet/guides/aspnet/performance/instrumentation/automatic-instrumentation/#sqlclient-integration) to generate spans for SQL operations. I'm calling both `o.AddAspNet(RequestSize.Always)` and `o.AddDiagnosticSourceIntegration();` (see the attached `SentrySdk.Init()` call below).

An action method in my controller throws a `System.Data.SqlClient.SqlException` when calling `SqlCommand.ExecuteReader()`. The error event shows up in Sentry but there is only an `http.server` span in the trace and no spans related to SQL (see attached image).

I have tried switching from `System.Data.SqlClient` to `Microsoft.Data.SqlClient` but nothing changed.

What do I need to do to make the SqlClient integration work correctly? Thank you!

Referenced Sentry packages:
```
Sentry.3.21.0
Sentry.AspNet.3.21.0
Sentry.DiagnosticSource.3.21.0
```

Excerpt from `Application_Start()` in my `Global.asax.cs`
```csharp
_sentry = SentrySdk.Init(o =>
{
    o.Dsn = Config.SentryDsn;
    o.Environment = Config.EnvironmentName;
    o.SendDefaultPii = true;
    o.AddAspNet(RequestSize.Always);
    o.AddDiagnosticSourceIntegration();
    o.TracesSampleRate = Config.SentryTracesSampleRate; //set to 1.0
    o.BeforeSend = sentryEvent =>
    {
        var httpCode = (sentryEvent.Exception as HttpException)?.GetHttpCode();
        if (httpCode == 404)
        {
            sentryEvent.Level = SentryLevel.Info;
        }
        return sentryEvent;
    };
});
```
Recorded trace:
<img width="519" alt="image" src="https://user-images.githubusercontent.com/6144728/194259313-264441ef-1d5f-4b2a-b46c-68d8285d48a4.png">

**Comment by SimonCropp** [comment, marked as answer]
@nixxcz unfortunately we cant support DiagnosticSource integration for SqlClient on net48 because the underlying  SqlClient only supports DiagnosticSource  on netcore. I have raise a docs PR to clarify this https://github.com/getsentry/sentry-docs/pull/5611/files

**Reply by nixxcz** [reply]
Thank you for clarification! I would suggest removing mentions of SqlClient integration from the ASP.NET documentation section altogetner if it's only relevant to ASP.NET Core. AFAIK, you cannot have ASP.NET application targeting netcore.

**Reply by mattjohnsonpint** [reply]
Agreed.  Also, FYI - While SqlClient doesn't use DiagnosticSource on .NET Framework, I believe EF Core does.  EF Core 3.1 will work on .NET Framework.  We also have a separate integration for the non-core EF 6.

(Though I do recognize that many people just use SqlClient without EF.)

**Reply by mattjohnsonpint** [reply]
Also, I opened a feature request issue for us to evaluate an ETW-based listener for SqlClient on NetFx.  https://github.com/getsentry/sentry-dotnet/issues/1974

**Reply by nixxcz** [reply]
Thank you, Matt. Indeed, I use just SqlClient without EF. I would greatly benefit from an ETW listener, thanks for the request!
